staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -123,6 +127,9 @@ func (s *SpdyRoundTripper) TLSClientConfig() *tls.Config {
 
 // Dial implements k8s.io/apimachinery/pkg/util/net.Dialer.
 func (s *SpdyRoundTripper) Dial(req *http.Request) (net.Conn, error) {
+	if s.UpgradeTransport != nil {
+		return apiproxy.Dial(req, s.UpgradeTransport)
+	}
```

Comment:
I have a slightly different take which keeps the `apiproxy.DialURL` in the same place, since the `net.Conn` still needs to be updated with `req.Write(conn)` afterwards. Please let me know what you think.